thoth-app/src/component/admin.rs

```diff
@@ -76,15 +98,44 @@ impl Component for AdminComponent {
     }
 
     fn change(&mut self, props: Self::Properties) -> ShouldRender {
+        let old_details = self.props.current_user.clone();
         self.props = props;
         if self.props.current_user.is_none() {
             self.link.send_message(Msg::RedirectToLogin);
+        } else {
+            let new_permissions = self
+                .props
+                .current_user
+                .as_ref()
+                .unwrap()
+                .resource_access
+                .clone();
+            // Raise an error if user's permission set is empty,
+            // but avoid raising repeated errors if permissions are unchanged
+            if new_permissions.restricted_to() == Some(vec![])
+                && (old_details.is_none()
+                    || (old_details.as_ref().unwrap().resource_access != new_permissions))
+            {
+                self.notification_bus.send(Request::NotificationBusMsg((
+                    PERMISSIONS_ERROR.into(),
+                    NotificationStatus::Danger,
+                )));
+            }
         }
         true
     }
 
     fn view(&self) -> Html {
-        if self.props.current_user.is_some() {
+        if self.props.current_user.is_some()
+            && self
```

Comment:
suggest checking the value rather than comparing it:

```
!self.props.current_user.as_ref().unwrap().resource_access.restricted_to().unwrap_or_default().is_empty()
```

Reply:
I don't think this will work - in the case where the user is a superuser, `restricted_to()` will return `None`, `unwrap_or_default()` will return `[]`, and the `is_empty()` check will return `True`, hiding everything from the user instead of showing everything.

Reply:
ah yes, I meant replacing the second part of the statement only.

```
if self.props.current_user.is_some() && !self.props.current_user.as_ref().unwrap().resource_access.restricted_to().unwrap_or_default().is_empty()
```

Reply:
No worries, that was clear - but same problem.

Reply:
you're absolutely right